### PR TITLE
feat: update docs bases on cancelled `prepare-2` PR

### PR DESCRIPTION
**Pull request by mdjastrzebski**
### Summary

Moved documentation and warning messages changes from feat: Prepare for 2.x release #266

### CODE REVIEW

File: src/helpers/errors.js [hunk 1]
```diff
@@ -60,9 +60,10 @@ export function printUnsafeWarning(functionName: string) {
   }
 
   console.warn(`
-  Deprecation Warning:
-  ${functionName} is not recommended for use and has been renamed to UNSAFE_${functionName}.
-  In react-native-testing-library 2.x only the UNSAFE_${functionName} name will work.
+    Deprecation Warning:
+    Warning:
+    ${functionName} promotes testing implementation details and is not recommended.
+    It may be removed in the future. Please test observable outcomes of rendering components.
```
Comment (thymikee) [inline src/helpers/errors.js:66]: Do we still need to yell at users for using `UNSAFE_` functions now? I think the prefix is enough to realize there's something fishy about these APIs

Comment (mdjastrzebski) [reply src/helpers/errors.js:66]: This is no longer called in UNSAFE_. We can remove this helper. What's your take?

Comment (mdjastrzebski) [reply src/helpers/errors.js:66]: Removed remaining warnFn invocations in #334

Comment (thymikee) [reply src/helpers/errors.js:66]: Let's do it

**Comment by thymikee** [comment]
There are some merge conflicts

File: src/helpers/errors.js (hunk 1 above)
Comment (mdjastrzebski) [reply src/helpers/errors.js]: I've removed the helper

**Comment by mdjastrzebski** [comment]
Fixed merge conflicts

**Pull request merged**